### PR TITLE
[#1467] fix(web): Fix refresh pages in a 404 error

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?

Resolve the issue where refreshing other pages results in a 404 error.

### Why are the changes needed?

Fix: #1467

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by yuqi1129** [comment]
@ch3yne 
Please resolve the conflict here.

File: server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
```
Comment (jerryshao) [inline server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java:2]: Can you please change to 2024.

File: server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class WebUIFilter implements Filter {
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) request;
+    String path = httpRequest.getRequestURI();
+    if (path.equals("/")) {
+      // Redirect to the index page.
+      httpRequest.getRequestDispatcher("/ui/index.html").forward(request, response);
+    } else if (path.startsWith("/ui/") && !path.contains(".")) {
```
Comment (jerryshao) [inline server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java:27]: What is the meaning of `!path.contains(".")`?

Comment (jerryshao) [reply server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java:27]: Do we have a scenario like "/ui/xxx/", which is ended with "/"?

Comment (xunliu) [reply server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java:27]: > What is the meaning of !path.contains(".")?
Skips `*.js`, `*.css`, and more.

**Review by jerryshao — APPROVED** [review]

**Comment by jerryshao** [comment]
@ch3yne can you please verify it locally?

**Pull request merged**